File: src/main/java/commands/slashadapters/adapters/SayAdapter.java
package commands.slashadapters.adapters;

import commands.runnables.gimmickscategory.SayCommand;
import commands.slashadapters.Slash;
import commands.slashadapters.SlashAdapter;
import commands.slashadapters.SlashMeta;
import net.dv8tion.jda.api.events.interaction.SlashCommandEvent;
import net.dv8tion.jda.api.interactions.commands.OptionType;
import net.dv8tion.jda.api.interactions.commands.build.CommandData;

@Slash(command = SayCommand.class)
public class SayAdapter extends SlashAdapter {

    public CommandData addOptions(CommandData commandData) {
        return commandData
                .addOption(OptionType.STRING, "text", "The text you want Lawliet to say", true);
    }

    @Override
    public SlashMeta process(SlashCommandEvent event) {
        return new SlashMeta(SayCommand.class, collectArgs(event));
    }

}

File: src/main/java/commands/slashadapters/adapters/SayAdapter.kt
package commands.slashadapters.adapters

import commands.runnables.gimmickscategory.SayCommand
import commands.slashadapters.Slash
import commands.slashadapters.SlashAdapter
import commands.slashadapters.SlashMeta
import mysql.hibernate.entity.guild.GuildEntity
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent
import net.dv8tion.jda.api.interactions.commands.OptionType
import net.dv8tion.jda.api.interactions.commands.build.SlashCommandData

@Slash(command = SayCommand::class)
class SayAdapter : SlashAdapter() {

    public override fun addOptions(commandData: SlashCommandData): SlashCommandData {
        return commandData.addOptions(
            generateOptionData(OptionType.STRING, "text", "say_text", true),
            generateOptionData(OptionType.CHANNEL, "channel", "say_channel", false)
        )
    }

    override fun process(event: SlashCommandInteractionEvent, guildEntity: GuildEntity): SlashMeta {
        return SlashMeta(SayCommand::class.java, collectArgs(event))
    }

}